### ISSUE TITLE
[Bug]: v0.20.0 Beta (.deb for Ubuntu 20.04) near to unresponsive app with two or more windows

### ISSUE THREAD

**Issue by RevAngel7** [FreeTubeApp/FreeTube#4875]
### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have encountered this bug in the [official downloads of FreeTube](https://github.com/FreeTubeApp/FreeTube#official-downloads).
- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the bug report I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [X] This issue contains only one bug.

### Describe the bug

1. Click one video with middle mouse button (opens second window with that video)
2. You can even stop that video
3. Switch back to main subscriptions page
4. Scroll down
-> Scrolling stops as if the application is using one core of my machine and as if that core would be at 100%

repeatable every time

### Expected Behavior

In previous versions the behaviour of multiple windows of freetube was no issue. Now even a second window with a stopped video brings every other window to near non responsiveness. 

Tested on 6 cores, 12 threads possible, max. freq. 4.5GHz

Sometimes freetube crashes if this overload occurs.

### Issue Labels

accessibility issue, causes crash, content not loading, usability issue

### FreeTube Version

v0.20.0 Beta (.deb for Ubuntu 20.04)

### Operating System Version

Ubuntu 20.04, kernel 6.8.1 mainline, mesa 24.1

### Installation Method

.deb

### Primary API used

Local API

### Last Known Working FreeTube Version (If Any)

V19

### Nightly Build

- [ ] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by RevAngel7** [comment]
The crashes actually got worse today. I am unable to watch 3 or more videos in a second video because of it.

**Comment by daiyam** [comment]
I was wondering if you have seen an increase of the CPU usage?

**Comment by RevAngel7** [comment]
I checked that yesterday. No. Still 1 core about 20%. But I had a change of kernel here from 6.8.1 to 6.8.4 and the situation improved somewhat. Maybe it was a scheduler issue? But then I asked myself, why just on one application.
Right now it seems that there is a initial delay, but not a constant delay any more. I can open a video, then switch to the main subscription list and scroll again, or at least try, but with aprox. 2 seconds delay I can scroll down without any more "loading" delays until the "load more videos" button.
It is a weird issue, I understand that, and if not more users have it that could mean it is on my systems side.
So I recommend putting a pin into it and wait. I keep an eye on it on my system and we will see if other users turn up here. If not, this issue could probably be closed  in 4 weeks time.

Does that sound reasonable to you @absidue ?

**Comment by absidue** [comment]
Just to be clear, me adding the https://github.com/FreeTubeApp/FreeTube/labels/OS%3A%20linux label, isn't me dismissing your issue, I'm just indicating that it seems to be a Linux only issue, as it doesn't happen on Windows.

**Comment by RevAngel7** [comment]
Well, I did a complete system reinstall. It is Ubuntu 24.04 LTS now. Standard Kernel (so standard drivers), nothing fancy or custom. That should have fixed things? Sadly not. The crashes of the whole app when closing a video window are the most annoying thing right now. Tested some alternative USB audio solutions, just to be sure that it is not related to that. No change. But why the hell does it look like I am the only user on linux with that issue? Video codecs work fine on MPV, VLC, Kodi - H.264, H.265 (HEVC), AV1, no issues there. Without and with a wireguard VPN - no difference here. I even switched the SSD for issues SMART could not detect. I don't know.

**Comment by RevAngel7** [comment]
This issue is still present, even tho the performance is better under ubuntu 24.04, the crashes still happen.

**Comment by RevAngel7** [comment]
Most times an app crash happens, when I close a second (video) window, no matter if it still plays or has finished playing.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Does this issue still persist for you on v0.21.0?

**Comment by RevAngel7** [comment]
Installing now and testing it as soon I have some time window for it.

**Comment by RevAngel7** [comment]
Preliminary testing of version v0.21.0 (.deb x64) with around 20 videos so far showed the issue gone. Response times between window changes (by middle mouse clicking videos) and closing video-windows mid-play and returning to the previous video-lists showed better response and no crashes so far.

It might be a little early to close this issue, but I keep using (and therefore testing) FreeTube regularly and will provide a conclusion after I watched more videos.

But until now it looks good!

**Comment by RevAngel7** [comment]
After more comprehensive tests I can confirm the issue GONE. Therefore this issue can be closed.

Thank you for improving this app to a real nice and well performing experience again! Thank you to anyone involved!
